src/journal/journald-server.c

```diff
@@ -2001,6 +2014,28 @@ static int vl_method_relinquish_var(Varlink *link, JsonVariant *parameters, Varl
         return varlink_reply(link, NULL);
 }
 
+static int vl_connect(VarlinkServer *server, Varlink *link, void *userdata) {
+        Server *s = userdata;
+
+        assert(server);
+        assert(link);
+        assert(s);
+
+        (void) server_start_idle_timer(s); /* maybe we are no longer idle */
```

Comment:
I don't understand this. Wouldn't we want to stay around as long as the connection is active?

Reply:
We do. The server_start_idle_timer() call will free the idle timer if the active connection counter is > 0. So we call the function here to free it, not to allocate this. The comment was supposed to make this clear.

I'll rename the function to server_start_or_stop_idle_timer() in the hope this makes this clearer?